pkg/server/plugin/upstreamauthority/wrapper.go

```diff
@@ -0,0 +1,80 @@
+package upstreamauthority
+
+import (
+	"context"
+	"crypto/x509"
+
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamca"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type wrapper struct {
+	upstreamCA upstreamca.UpstreamCA
```

Comment:
Is there any reason to not use a *upstreamca.UpstreamCA instead?

Reply:
*upstreamca.UpstreamCA is used on catalog to  indicate to Fill that the plugin is optional, here it is not optional it is somethign we required

Reply:
Sorry, wasn't my intention to use a pointer to an interface. Resolving..